src/utils/image.js

```diff
@@ -651,6 +651,100 @@ export class RawImage {
         }
     }
 
+    /**
+     * Generates a 1D Gaussian kernel.
+     * @param {number} kernelSize - Kernel size (must be odd).
+     * @param {number} sigma - Standard deviation of the Gaussian.
+     * @returns {Array<number>} - The 1D Gaussian kernel.
+     */
+    generateGaussianKernel(kernelSize, sigma) {
+        // Kernel must be odd because each pixel must sit evenly in the middle.
+        if (kernelSize % 2 === 0) {
+            throw new Error('Kernel size must be odd.');
+        }
```

Comment:
The reason for this, is because a kernel is essentially a square that is applied to every pixel in an image.

Consider a grid of 5x5, it would look like this, with our pixel (`P`) in the middle.
```
. . . . .
. . . . .
. . P . .
. . . . .
. . . . .
```
Here, it is clear to see that there is an even spacing around the pixel.

Now, consider what would happen if we used an even kernel size of `4`.
```
. . . .
. . . .
. . P .
. . . .
```
The pixel does not sit neatly in the middle of the kernel and this is a problem.